urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/grinder/client/LogFiler.java

```diff
@@ -0,0 +1,143 @@
+package grinder.client;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Timestamp;
+import java.util.UUID;
+
+/**
+ * The class LogFiler provides a simple way of generating logfiles.
+ */
+public class LogFiler {
+
+	/** The log file name. */
+	private final String fileName;
+
+	/** The BufferedWriter. */
+	private BufferedWriter bw = null;
+
+	/** Reference to the client. */
+	private final Client client;
+
+	/**
+	 * Instantiates a new log file handler.
+	 *
+	 * @param fileName the logfile name
+	 * @throws IOException
+	 */
+	public LogFiler(final Client c, final String fileName) throws IOException {
+		client = c;
+		this.fileName = fileName;
+
+		bw = new BufferedWriter(new FileWriter(fileName));
+	}
+
+	/**
+	 * Write a grinder message to the log.
+	 *
+	 * @param clientID the client id
+	 * @param content the preformatted content string
+	 * @see logfileformat.txt
+	 */
+	public synchronized void log(final UUID clientID, final String content) {
+
+		final Timestamp t = new Timestamp(System.currentTimeMillis());
+
+		if (bw != null)
+		{
+			try
+			{
+				bw.write(clientID.toString() + "|" + t + "|" + content);
+				bw.newLine();
+				bw.flush();
+
+			} catch (final IOException e)
+			{
+				client.cgui.logMessage("LogFiler BufferedWriter IO Error! " + e);
+			}
+		}
+	}
+
+	public void closeLogFile() {
+		// shutdown writer
+		try
+		{
+			if (bw != null)
+			{
+				bw.flush();
+				bw.close();
+				bw = null;
+			}
+		} catch (final IOException ex)
+		{
+			client.cgui.logMessage("Cannot close log file! " + ex);
+		}
+	}
+
+	/**
+	 * Gets the file.
+	 *
+	 * @return the file
+	 */
+	public File getFile() {
+		return new File(fileName);
+	}
+
+	/**
+	 * Gets the logfile name
+	 *
+	 * @return the file
+	 */
+	public String getFilename() {
+		return fileName;
+	}
+
+	/**
+	 * Delete log file.
+	 */
+	public void deleteLogFile() {
+		(new File(getFilename())).delete();
+	}
+
+	/**
+	 * Gets the bytes from a file.
+	 *
+	 * @param file the file
+	 * @return the bytes from file
+	 * @throws IOException
+	 */
+	public byte[] getFileByteArray() throws IOException {
+
+		final File file = new File(fileName);
+
+		final InputStream is = new FileInputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *is*. The resource is referenced at line 129. The resource is closed at line 139. There are other execution paths that do not contain closure statements, for example, when an if condition at line 135 evaluates to true. Ensure *is* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).